app/dashboard/router.py

```diff
@@ -194,17 +199,97 @@ def update(self, validated_data):
         return bounty
 
 
+class HackathonProjectFeedbackSerializer(serializers.ModelSerializer):
+
+    sender = ProfileSerializer()
+
+    class Meta:
+        model = HackathonProjectFeedbackEntry
+        fields = ('pk', 'sender', 'comment', 'rating')
+
 class HackathonProjectSerializer(serializers.ModelSerializer):
     bounty = BountySerializer()
     profiles = ProfileSerializer(many=True)
     hackathon = HackathonEventSerializer()
+    project_feedbacks_got = HackathonProjectFeedbackSerializer(many=True)
+    avg_rating = serializers.JSONField(source='get_avg_rating')
 
     class Meta:
         model = HackathonProject
-        fields = ('pk', 'chat_channel_id', 'status', 'badge', 'bounty', 'name', 'summary', 'work_url', 'profiles', 'hackathon', 'summary', 'logo', 'message', 'looking_members')
+        fields = ('pk', 'chat_channel_id', 'status', 'badge', 'bounty', 'name', 'summary', 'work_url', 'profiles', 'hackathon', 'summary', 'logo', 'message', 'looking_members', 'project_feedbacks_got', 'avg_rating', 'winner', 'admin_url')
         depth = 1
 
 
+class HackathonProjectsPagination(PageNumberPagination):
+    page_size = 100
+
+
+class HackathonProjectsViewSet(viewsets.ModelViewSet):
+    queryset = HackathonProject.objects.prefetch_related('bounty', 'profiles').all().order_by('id')
+    serializer_class = HackathonProjectSerializer
+    pagination_class = HackathonProjectsPagination
+
+    def get_queryset(self):
+
+        q = self.request.query_params.get('search', '')
+        order_by = self.request.query_params.get('order_by', '-created_on')
+        skills = self.request.query_params.get('skills', '')
+        filters = self.request.query_params.get('filters', '')
+        sponsor = self.request.query_params.get('sponsor', '')
+        hackathon_id = self.request.query_params.get('hackathon', '')
+        rating = self.request.query_params.get('rating', '')
+
+        if hackathon_id:
+            try:
+                hackathon_event = HackathonEvent.objects.get(id=hackathon_id)
+            except HackathonEvent.DoesNotExist:
+                hackathon_event = HackathonEvent.objects.last()
+
+            queryset = HackathonProject.objects.filter(hackathon=hackathon_event).exclude(
+                status='invalid').prefetch_related('profiles', 'bounty').order_by(order_by, 'id')
+
+            if sponsor:
+                queryset = queryset.filter(
+                    Q(bounty__github_url__icontains=sponsor) | Q(bounty__bounty_owner_github_username__in=[sponsor])
```

Comment:
why use the `__in` filter instead of icontains or equals?

Reply:
good question, was just migrated from the other function as is.